core/src/main/java/org/renjin/primitives/text/Text.java

```diff
@@ -203,6 +203,25 @@ public static void bindtextdomain(String domain, String dirname) {
     // translation not yet supported.
   }
 
+  @Internal
+  public static StringVector enc2utf8(SEXP inputVector) {
+    if (!(inputVector instanceof StringVector)) {

```

Comment:
Use the annotations/method signature to avoid these types of checks. Just declare the method as 

```.java
public static StringVector enc2utf8(StringVector inputVector)
```

And the code generate will do the type checking for you.